app/js/services/situationService.js

```diff
@@ -226,6 +223,19 @@ angular.module('ddsApp').factory('SituationService', function($http, $sessionSto
                 .concat(enfants)
                 .concat(individus.slice(1));
             situation.individus = individus;
+        },
+
+        ressourcesYearMoins2Captured: function(situation) {
+            var ressourcesYM2Names = ['rncRevenusActivite', 'rncAutresRevenus', 'rncPensionsRetraitesRentes', 'fraisReelsDeductibles', 'rncPensionsAlimentaires', 'rncPensionsAlimentairesVersees'];
+            var ressourcesYM2 = [].concat.apply(
+                [],
+                situation.individus.map(function(individu) {
+                    return individu.ressources.filter(function(ressource) {
+                        return ressourcesYM2Names.indexOf(ressource.type) > 0;
+                    });
+                })
+            );
+            return ressourcesYM2.length > 0 || situation.rfr !== undefined;
```

Comment:
Make the righthand condition a possible early exit at the beginning of the function to avoid a costly multiple iterations if `rfr !== undefined`.